crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -57,17 +67,84 @@ where
         self.inner.install_filter(FilterKind::PendingTransaction).await
     }
 
-    async fn filter_changes(&self, _id: SubscriptionId<'_>) -> RpcResult<FilterChanges> {
-        todo!()
+    async fn filter_changes(&self, id: SubscriptionId<'_>) -> RpcResult<FilterChanges> {
+        let id = id.into_owned();
+        let info = self.inner.client.chain_info().to_rpc_result()?;
+        let best_number = info.best_number;
+
+        let (start_block, kind) = {
+            let mut filters = self.inner.active_filters.inner.lock().await;
```

Comment:
This seems a bit dangerous if some client is spamming is with get filter changes requests? if we have all the filters in one lock, then a client can perma lock resources here?

Reply:
this is an async mutex and we only hold it for as long as we need to update the filter params.